pkg/operator/status.go

```diff
@@ -65,6 +65,13 @@ func setAvailableFalse(operatorConfig *operatorv1.ServiceCA, reason, message str
 	})
 }
 
+func setUpgradeableTrue(operatorConfig *operatorv1.ServiceCA) {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+		Type:   operatorv1.OperatorStatusTypeUpgradeable,
+		Status: operatorv1.ConditionTrue,
```

Comment:
Can we set `Reason: "AsExpected"` or some such, because it makes it easier to distinguish between "we're happy" and "oops, we forgot to explain why we were sad" in Telemetry like `count by (name,reason) (cluster_operator_conditions)`?

Reply:
Thanks, updated. I couldn’t find a Go constant for `"AsExpected"`, using a string like library-go and others.

Reply:
hmm, you pass `reason` into `setUpgradeableTrue` now (good), but I don't see you actually passing it through to `SetOperatorCondition`?

Reply:
Oops, stupid of me. Thanks, hopefully fixed, and I’ll actually test this in practice to confirm this time.

Reply:
`managementState: Managed`:
```console
$ oc get serviceca/cluster -o yaml
…
status:
  conditions:
  - lastTransitionTime: "2021-04-14T11:36:15Z"
    reason: OperatorSyncLoopComplete
    status: "False"
    type: Degraded
  - lastTransitionTime: "2021-04-14T11:44:51Z"
    message: All service-ca-operator deployments updated
    reason: ManagedDeploymentsCompleteAndUpdated
    status: "False"
    type: Progressing
  - lastTransitionTime: "2021-04-14T11:44:43Z"
    reason: ManagedDeploymentsCompleteAndUpdated
    status: "True"
    type: Available
  - lastTransitionTime: "2021-04-14T11:36:15Z"
    reason: AsExpected
    status: "True"
    type: Upgradeable
  - lastTransitionTime: "2021-04-14T11:36:17Z"
    status: "False"
    type: ResourceSyncControllerDegraded
…
$ oc get clusteroperators/service-ca -o yaml
…
status:
  conditions:
  - lastTransitionTime: "2021-04-14T11:36:10Z"
    message: All is well
    reason: AsExpected
    status: "False"
    type: Degraded
  - lastTransitionTime: "2021-04-14T11:44:51Z"
    message: 'Progressing: All service-ca-operator deployments updated'
    reason: AsExpected
    status: "False"
    type: Progressing
  - lastTransitionTime: "2021-04-14T11:36:15Z"
    message: All is well
    reason: AsExpected
    status: "True"
    type: Available
  - lastTransitionTime: "2021-04-14T11:36:15Z"
    message: All is well
    reason: AsExpected
    status: "True"
    type: Upgradeable
…
```

and `managementState: Unmanaged`:
```console
$ oc get serviceca/cluster -o yaml
…
status:
  conditions:
  - lastTransitionTime: "2021-04-14T11:36:15Z"
    reason: OperatorSyncLoopComplete
    status: "False"
    type: Degraded
  - lastTransitionTime: "2021-04-14T11:44:51Z"
    message: All service-ca-operator deployments updated
    reason: ManagedDeploymentsCompleteAndUpdated
    status: "False"
    type: Progressing
  - lastTransitionTime: "2021-04-14T11:44:43Z"
    reason: ManagedDeploymentsCompleteAndUpdated
    status: "True"
    type: Available
  - lastTransitionTime: "2021-04-14T11:36:15Z"
    reason: AsExpected
    status: "True"
    type: Upgradeable
  - lastTransitionTime: "2021-04-14T11:36:17Z"
    status: "False"
    type: ResourceSyncControllerDegraded
…
$ oc get clusteroperators/service-ca -o yaml
…
status:
  conditions:
  - lastTransitionTime: "2021-04-14T12:03:47Z"
    reason: Unmanaged
    status: Unknown
    type: Available
  - lastTransitionTime: "2021-04-14T12:03:47Z"
    reason: Unmanaged
    status: Unknown
    type: Progressing
  - lastTransitionTime: "2021-04-14T12:03:47Z"
    reason: Unmanaged
    status: Unknown
    type: Degraded
  - lastTransitionTime: "2021-04-14T12:03:47Z"
    reason: Unmanaged
    status: Unknown
    type: Upgradeable
…
```